homeassistant/components/hassio/update.py

```diff
@@ -130,27 +126,15 @@ def entity_picture(self) -> str | None:
             return f"/api/hassio/addons/{self._addon_slug}/icon"
         return None
 
-    def _strip_release_notes(self) -> str | None:
-        """Strip the release notes to contain the needed sections."""
-        if (notes := self._addon_data[ATTR_CHANGELOG]) is None:
-            return None
-
-        if (
-            f"# {self.latest_version}" in notes
-            and f"# {self.installed_version}" in notes
-        ):
-            # Split the release notes to only what is between the versions if we can
-            new_notes = notes.split(f"# {self.installed_version}")[0]
-            if f"# {self.latest_version}" in new_notes:
-                # Make sure the latest version is still there.
-                # This can be False if the order of the release notes are not correct
-                # In that case we just return the whole release notes
-                return new_notes
-        return notes
-
     async def async_release_notes(self) -> str | None:
         """Return the release notes for the update."""
-        return self._strip_release_notes()
+        changelog = await self.coordinator.get_changelog(self._addon_slug)
+        regex_pattern = re.compile(
```

Comment:
I'd make the compiled regex a constant to avoid compiling it every time

Reply:
It does include the versions as variables, so it can't be constant.

We could cache the regex though.

FWIW, if pre-compiled, the regex method is pretty much the same speed as the previous version. The advantage is that it cuts the release notes slightly cleaner (avoids left-over pound signs).

Reply:
Its not called very often so unlikely to make a difference or worth the complexity of trying to cache it.